### ISSUE TITLE
Manuscript–code consistency check: response & fix-tracking (2026-08-25)

### ISSUE THREAD

**Issue by YonganZhang** [YonganZhang/FMS-KAN-well-logging#1]
# 关于 `manuscript_consistency_check.md` 的回复与处理

**日期：** 2026-08-25
**回应对象：** 齐振鹏《论文与最新 GitHub 代码一致性检查》(2026-08-25)
**仓库：** <https://github.com/YonganZhang/FMS-KAN-well-logging>
**本轮修复提交：** `d31b236`

---

## 一、总体结论

这份一致性检查做得很扎实,我们逐条一线核实(读了 `finalize_pipeline.py`、`fms_kan.py`、`extract3.py`、`regenerate_v2.py`、公开 `data/` 与 README),**报告里的条目基本条条成立**,先谢谢齐振鹏的严谨。

需要先讲清楚**问题的性质**,避免误解:

- **主结果不是造假。** R²、模型、对比数字都是在真实数据上真跑出来的;RF sanity 断言也在保护主口径。
- 问题分两类:
  1. **大部分是"代码没上传好 / 口径没梳理清楚"** —— 东西真做了,只是打包、标注、叙事没对齐(复现性、逐井口径、多尺度措辞、消融配置、图件术语)。
  2. **一处是"方法写多了"(over-claim)** —— `XGBoost–SHAP 特征引导差异化配置边级网格` 这个机制,重建代码里**没有实现**(实际用的是随机森林特征重要性做变量分析)。这不是伪造数据,但属于方法描述超过实现,必须诚实处理。

---

## 二、逐条回应

| # | 齐的条目 | 性质 | 本轮处理 | 状态 |
|---|---|---|---|---|
| 1 | 多尺度 `5→10→20`+softmax 与代码 `5→10→15` 细化不一致 | 叙事没对齐实现(代码真做的是逐级 refinement) | 中文稿上轮已改为 `5→10→15` 逐级细化、删 softmax;本轮改 `fms_kan.py` docstring、README、架构图文字 | 中文稿✓ 代码✓ / 框架图重画+英文稿**待办** |
| 2 | XGBoost–SHAP 特征引导未在管线实现 | **over-claim**(核心创新之一代码没做) | README 已诚实降级为"随机森林特征重要性分析,用于变量选择与解释" | README✓ / **论文正文降级 or 真做，待定方向** |
| 3 | 公开副本不能运行(脚本读不存在的 `clean/ZXX*_clean.csv`) | 纯打包没接好 | 新增 `code/prepare_clean.py`,从公开 `data/Well*.csv` 桥接出 `code/clean/`,已验证公开副本可跑(5812 行、各目标样本充足) | **✓ 已修并上传** |
| 4 | 消融"同配置"比较未实现(标准KAN 12特征/Adam vs FMS-KAN 18特征/LBFGS) | 实验设计非配对(数字真,但增益归因不严) | 暂缓 | **待定:补同配置标准KAN重训 or 收紧表述** |
| 5 | "逐井 test-set"实为全井样本预测 | 口径标错(数字真实,标签错) | README 已加 caveat 说明 `r2_per_well` 是逐井重建、非逐井留出测试 | README✓ / **论文表2+图口径待改 or 重导 test-only** |
| 6 | 特征重要性文字与热图不符;图只 12 原始字段(非18) | 图文不符 | 暂缓 | **待办:重画热图(标注 RF+字段范围)、正文描述对齐图** |
| 7 | 框架图输入字段与现稿不符(BRIT/PV/PF/POR/PERM 等) | 框架图用了旧字段 | 暂缓 | **待办:框架图重画(实际 12+6 派生特征、去 softmax)** |
| 8 | 图件术语/单位/面板标签不统一(Measured/σH/K/WellA) | 图件规范 | 暂缓 | **待办:统一术语、单位、面板标签** |
| 暂缓a | 三条公式占位符 | 已用代码真跑填入 | `extract3.py` 真跑:SHMAX/TOC 填入代码提取的解析式(符号化 R²≈0.84/0.83,如实标注精度损失);PERM 符号化 R²<0 → **诚实降级、不给公式、不 AI 编** | 中文稿✓ |
| 暂缓b | 消融均值 0.004 | 由上两行原始精度相减 | 保留,投稿前给未取整可核验值 | 待补 |
| 暂缓c | 目标期刊字段 | 待选刊 | — | 待定 |

---

## 三、本轮已完成并上传(提交 `d31b236`)

1. **复现性桥接** `code/prepare_clean.py`:公开副本 `python code/prepare_clean.py` 后即可跑 `train_pooled.py` / `finalize_pipeline.py` / `extract3.py`(在甲方红线内,仅用已发布 50% 脱敏数据)。
2. **README 诚实化**:
   - 多尺度更正为 `G=5→10→15` 逐级 refinement(非 softmax 融合);
   - "XGBoost–SHAP 边级网格预算" → 降级为实际实现的**随机森林特征重要性分析**;
   - 明确区分**连续模型 R²=0.997** 与**符号公式 R²≈0.84(SHMAX)/0.83(TOC)**,PERM 保留数值模型、不作公式;
   - 加逐井"重建 vs 留出测试"caveat;复现步骤接入 `prepare_clean`。
3. **代码注释** `fms_kan.py`:`5→10→20` → `5→10→15 refinement`。

---

## 四、需要龚老师 / 齐振鹏拍板的方向(定了我就按序改完)

- **① 特征引导**:(A) 真补 XGBoost→SHAP→边级网格配置+重训(工程量大,干净数据下增益可能很小);(B) 论文降级为"树模型特征重要性分析"(诚实、扛审,创新收缩到"多尺度+可解释")。**建议 B**,除非要保留该卖点。
- **② 消融**:(A) 补同配置标准KAN(18特征/同结构/同优化器)重训;(B) 收紧论文表述。**建议 A**(不难,能让消融站得住)。
- **③ 逐井口径**:(A) 按 70/15/15 索引重导 test-only 逐井;(B) 表述改为"全井重建剖面"。**建议 A**。
- **④ 英文稿**:目前仓库/项目里没有与最新中文稿同步的英文稿(报告依据的是 8-05 旧英文稿),最后统一生成 Elsevier 英文稿。

方向确定后,我会完成框架图重画、热图与正文特征描述对齐、图件术语统一、逐井口径与消融的补跑,以及英文稿同步。

---

*本回复同步发布为仓库 issue,便于团队跟踪各条处理状态。*

**Comment by YonganZhang** [comment]
## 修改稿已上传到仓库

最新论文(已完成本轮全部一致性修改)现在在仓库 `paper/` 目录:

- 📄 **PDF(可在线预览,12 页)**:[`paper/main_kan.pdf`](https://github.com/YonganZhang/FMS-KAN-well-logging/blob/master/paper/main_kan.pdf)
- 📝 **LaTeX 源码**:[`paper/main_kan.tex`](https://github.com/YonganZhang/FMS-KAN-well-logging/blob/master/paper/main_kan.tex)(附 `elsarticle.cls` 与 `figures/`,可直接 xelatex 编译)
- 📁 目录:https://github.com/YonganZhang/FMS-KAN-well-logging/tree/master/paper

这一版对应本 issue 与 #2 的全部修改:同一模型公式提取、配对多尺度消融、随机森林特征重要性(已删除 XGBoost–SHAP 特征引导)、12 个原始特征、逐井口径与措辞更正。师弟可据此核对与继续修改。

**Comment by YonganZhang** [comment]
## 全量数据 + 绘图数据已上传(可直接重画所有图)

为方便重绘图件,现已上传:

- 📊 **全量脱敏数据**(WellA–D,全部深度样本、深度连续;已去除内部井号与 BRITL,仅 DEPTH + 12 原始测井特征 + 3 目标):[`data/`](https://github.com/YonganZhang/FMS-KAN-well-logging/tree/master/data)
- 🖼️ **现成绘图数据**(每口井每目标的 `depth, measured, predicted`):[`code/data_desensitized_v2/`](https://github.com/YonganZhang/FMS-KAN-well-logging/tree/master/code/data_desensitized_v2) + `code/perwell_all.csv`

### 重画所有图(4 类:散点 / 深度剖面 / 提升热图 / 特征重要性)

```bash
python code/prepare_clean.py     # 全量 data/ -> code/clean/(现为全数据)
python code/regenerate_v2.py     # 用 data_desensitized_v2 + perwell_all + clean 出全部图
```

- `regenerate_v2.py` 直接读上面这些数据,不必重跑 KAN。
- 若要换模型/公式后重出预测,再跑一遍 `finalize_pipeline.py` 即可刷新 `data_desensitized_v2/`。

### 说明

- `data/` 现在是**全量**数据(11,621 行),README 的数据伦理段/样本数/快速开始已同步改为"全量脱敏数据集",不再是此前的 50% 子集。
- 数据已脱敏(WellA–D、无内部井号),按甲方授权公开。
